test/build-test/prepublish/lib/checkNpmAuth-spec.js

```diff
@@ -0,0 +1,67 @@
+const sinon = require( 'sinon' );
+const chai = require( 'chai' );
+const expect = chai.expect;
+const sinonChai = require( 'sinon-chai' );
+chai.use( sinonChai );
+const proc = require( 'child-process-promise' );
+const path = require( 'path' );
+const rootDir = require( '../root-path' );
+const libDir = path.join( rootDir, 'scripts', 'npm', 'prepublish', 'lib' );
+const util = require( libDir );
+const printLn = util.printLn;
+let execStub;
+let processStub;
+let printErrorStub;
+let printInfoStub;
+
+describe( 'checkNpmAuth', () => {
+  beforeEach( () => {
+    execStub = sinon.stub( proc, 'exec' );
+    processStub = sinon.stub( process, 'exit' );
+    printErrorStub = sinon.stub( printLn, 'error' );
+    // Stubbing printLn.info avoids printing the user name mid-test
+    printInfoStub = sinon.stub( printLn, 'info' );
+  } );
+
+  afterEach( () => {
+    execStub.restore();
+    processStub.restore();
+    printErrorStub.restore();
+    printInfoStub.restore();
+  } );
+
+  it( 'should warn you if you’re not logged into NPM', () => {
+    // Stubbing console.error avoids printing the expected error mid-test
+    const consoleStub = sinon.stub( console, 'error' );
+    const errorMessage =
+      'You\'re not logged into npm. You need to authorize ' +
+      'this machine using `npm login`.';
+
+    execStub.withArgs( 'npm whoami' ).rejects( 'ENEEDAUTH' );
+
+    return util.checkNpmAuth( 'cf-core' ).then( () => {
+      expect( printErrorStub ).to.be.calledWith( errorMessage );
+      consoleStub.restore();
+    } );
+  } );
+
+  it( 'should exit if the user is not authorized on NPM', () => {
+    const errorMessage =
+      'You\'re not listed as a maintainer in npm. ' +
+      'Talk to a node-ledgable coworker for assistance.';
```

Comment:
![ba dum tss](https://media.giphy.com/media/9T3hk5HWKGdpK/giphy.gif)

Reply:
"ledgable"?

Reply:
@Scotchester know-ledgable

@anselmbradford wish I could take credit but that is all @contolini

Reply:
Woof.

Reply:
![nl80o](https://user-images.githubusercontent.com/1060248/34845351-98ad6438-f6e2-11e7-8b02-ade321bc7b47.gif)